### ISSUE TITLE
Strict Mode CleartextNetworkViolation: Detected cleartext network traffic

### ISSUE THREAD

**Issue by leinardi** [firebase/firebase-android-sdk#3158]
### [READ] Step 1: Are you in the right place?

I think so.

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: Android Studio Arctic Fox | 2020.3.1 Patch 3
  * Firebase Component: Crashlytics
  * Component version: 18.2.4

### [REQUIRED] Step 3: Describe the problem
I just added Crashlytics to a brand new project and, after enabling it, Strict Mode throws immediately a CleartextNetworkViolation:

```
2021-11-21 17:48:07.190 13126-13141/com.leinardi.forlago.debug D/StrictMode: StrictMode policy violation: android.os.strictmode.CleartextNetworkViolation: Detected cleartext network traffic from UID 10165 to /172.217.168.30x00000000 45 00 00 68 2A 66 40 00 40 06 AF 3D 0A 00 02 10 E..h*f@.@..=....
    0x00000010 AC D9 A8 03 86 04 01 BB BF 93 AC C9 3B A9 80 DD ............;...
    0x00000020 50 18 FF FF 61 47 00 00 14 03 03 00 01 01 17 03 P...aG..........
    0x00000030 03 00 35 F2 BB AD 0F 7D 88 72 8B 3A FE 6D E7 25 ..5....}.r.:.m.%
    0x00000040 C7 95 B2 43 00 87 9C 28 46 56 2D 47 3C 46 E5 30 ...C...(FV-G<F.0
    0x00000050 EE 73 11 D9 B6 71 07 F5 2D 49 97 61 F9 D0 A3 3C .s...q..-I.a...<
    0x00000060 08 73 9D 7E 7B 18 30 E3                         .s..{.0. 
        at android.os.StrictMode.onCleartextNetworkDetected(StrictMode.java:2246)
        at android.app.ActivityThread$ApplicationThread.notifyCleartextNetwork(ActivityThread.java:1764)
        at android.app.IApplicationThread$Stub.onTransact(IApplicationThread.java:1202)
        at android.os.Binder.execTransactInternal(Binder.java:1184)
        at android.os.Binder.execTransact(Binder.java:1143)
```
Immediately on startup of the app on API 28 and 31, and all is needed is to just add Crashlytics as dependency.
The project it's just a template with 2 screens and zero network calls. Removing Crashlytics as dependency removes also the violation.

#### Relevant Code:
A reproduction project is available here: https://github.com/leinardi/Forlago/tree/crashlytics
Crashlytics is added as dependency here: https://github.com/leinardi/Forlago/blob/crashlytics/modules/core-logging/build.gradle#L31
and as Gradle plugin here:
https://github.com/leinardi/Forlago/blob/crashlytics/build-conventions/src/main/groovy/forlago.android-app-conventions.gradle#L31
Strict Mode configuration is available here: https://github.com/leinardi/Forlago/blob/crashlytics/app/src/main/kotlin/com/leinardi/forlago/Forlago.kt#L63
#### Steps to reproduce:

1. Clone the repository https://github.com/leinardi/Forlago and checkout the branch `crashlytics`
2. Provide a valid `google-services.json` (otherwise Gradle won't load the Crashlytics plugin)
3. Run the Forlago app on an API 31 Emulator
4. Check the Logcat for `CleartextNetworkViolation`

Please let me know if I can help with more info or a remote debug session.

**Comment by argzdev** [comment]
Hi @leinardi, thanks for the detailed report. I tried to clone the project and provided a `google-services.json` as needed. However, I encountered a `Android App Versioning Gradle Plugin works with git tags but root project 'Forlago' is not a git root directory, and a valid gitRootDirectory is not provided.` issue. May I know what I'm missing to make this project work?

**Comment by argzdev** [comment]
In the meantime, I tried reproducing the issue by creating a new project with only Crashlytics plugin, google-services.json, and a custom Application class with the Strict Mode configuration, but I wasn't able to encounter the issue on API versions 28 or 31. With that said, it could be that the issue is pointing somewhere else. You could also try using our quickstart [guides](https://github.com/firebase/quickstart-android), then add Strict Mode configuration, this will help you cross reference with your app to better investigate the issue.

**Comment by leinardi** [comment]
From the error it seems that the [app versioning plugin](https://github.com/ReactiveCircus/app-versioning) can't recognize the project as a git project. I just pushed a new commit to disable it. If you do a git pull should be fine now.

**But** today I can't reproduce the issue anymore! Doesn't seem related to the app versioning plugin, because I could not reproduce it even before the above changes.

Yesterday the issue was 100% reproducible on the `crashlytics` branch, today I tried already several times on different API levels and I can't get it once. `master` branch is also fine...

Just for context, I created the Firebase project for the app yesterday and immediately I tried to setup Crashlytics. Could it be that the beck-end was not ready yet and the SDK was trying to reach some other URLs using HTTP protocol? I don't know, I'm just guessing. There are no changes on the code on my side...

I will close this issue and re-open it if I manage to reproduce it again.

Thanks a lot for taking the time to investigate it :+1:
